src/main/scala/gestalt/Toolbox.scala

```diff
@@ -2,253 +2,226 @@ package scala.gestalt
 
 import scala.collection.immutable.Seq
 
-trait Toolbox { t =>
-  // portable trees -- minimum assumptions
-  type Tree <: { def tpe: Type } // TODO: structural types performance penalty.
-  type TypeTree <: Tree          // safety by construction -- implementation can have TypeTree = Tree
-  type Type
+trait Modifiers[T] {
```

Comment:
I agree it's better to make modifiers immutable. Regarding the ease of usage, do you think it's more friendly to define some flags, so that they can easily set or unset some modifiers?

Initially, I made `Modifiers` inside `Toolbox`, but it seems to make the code inside toolbox more cluttered, so I decided to move it out. The abstract interface doesn't need to know anything about `T`, so I think it's probably fine to keep it outside of Toolbox.